File: 41-FirstMissingPositive/41-FirstMissingPositive.java
// Last updated: 9/26/2025, 2:06:56 PM
class Solution {
    public int firstMissingPositive(int[] nums) {
      HashSet<Integer> set=new HashSet<>();
      for(int i=0;i<nums.length;i++){
        if(nums[i]>0){
            if(!set.contains(nums[i])){
                set.add(nums[i]);
            }
        }
      }
      int i=1;
      while(true){
        if(!set.contains(i)) return i;
        i++;
      }
    }
}

File: 41-FirstMissingPositive/41-FirstMissingPositive.py
# Last updated: 8/1/2025, 10:41:56 PM
class Solution:
    def firstMissingPositive(self, nums: List[int]) -> int:
        num_set = set(nums)
        i = 1
        while i in num_set:
            i += 1
        return i